testament/testament.nim

```diff
@@ -79,6 +83,20 @@ type
     startTime: float
     debugInfo: string
 
+  CompilerOutput = object
+    ## Describes the output of a a compiler invocation.
+    cmd: string    ## the command that was run
+    nimout: string ## compiler output
+    output: string ## test output
+
+    # information related to reported compiler errors (unused if
+    # none were reported):
+    file, msg: string
+    line, column: int
+
+    debugInfo: seq[string] ## debug info to give more context
+    err: TResultEnum       ## the reason the test failed (or succeeded)
```

Comment:
This is going to be turned into a boolean as part of one of the following refactors.